lib/storage/storage.go

```diff
@@ -2819,6 +2821,48 @@ func (s *Storage) mustOpenIndexDBTables(path string) (next, curr, prev *indexDB)
 	return next, curr, prev
 }
 
+func (s *Storage) startReadonlyWatchdog() {
+	s.readonlyWatchdogWG.Add(1)
+	go func() {
+		s.readonlyWatchDog()
+		s.readonlyWatchdogWG.Done()
+	}()
+}
+
+func (s *Storage) readonlyWatchDog() {
+	timeout := 10 * time.Second
+
+	for {
+		timeoutChan := time.After(timeout)
+		tmpFileChan := make(chan *os.File, 1)
+		go func() {
+			tmpFile, err := os.CreateTemp(s.path, "is_readonly_watchdog")
+			if err != nil {
+				tmpFileChan <- nil
+				logger.Errorf("cannot write to the disk %s", err)
+			} else {
+				tmpFileChan <- tmpFile
+			}
+		}()
+
+		select {
+		case tmpFile := <-tmpFileChan:
+			if tmpFile == nil {
+				s.cannotWrite.Store(true)
+			} else {
+				tmpFile.Close()
```

Comment:
Before closing, a file write test should be performed and a `sync` flush should be executed to trigger disk write, such as writing a 10MB byte stream.

Reply:
`s.cannotWrite.Store(true)`  => `s.isReadOnly.Store(true)`